Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -1124,42 +1124,42 @@ void CompareWorkspaces::doPeaksComparison(PeaksWorkspace_sptr tws1,
       } else if (name == "l") {
         s1 = peak1.getL();
         s2 = peak2.getL();
-      } else if (name == "wavelength") {
+      } else if (name == "Wavelength") {
         s1 = peak1.getWavelength();
         s2 = peak2.getWavelength();
-      } else if (name == "energy") {
+      } else if (name == "Energy") {
         s1 = peak1.getInitialEnergy();
         s2 = peak2.getInitialEnergy();
-      } else if (name == "tof") {
+      } else if (name == "TOF") {
         s1 = peak1.getTOF();
         s2 = peak2.getTOF();
-      } else if (name == "dspacing") {
+      } else if (name == "DSpacing") {
         s1 = peak1.getDSpacing();
         s2 = peak2.getDSpacing();
-      } else if (name == "intens") {
+      } else if (name == "Intens") {
         s1 = peak1.getIntensity();
         s2 = peak2.getIntensity();
-      } else if (name == "sigint") {
+      } else if (name == "SigInt") {
         s1 = peak1.getSigmaIntensity();
         s2 = peak2.getSigmaIntensity();
-      } else if (name == "bincount") {
+      } else if (name == "BinCount") {
         s1 = peak1.getBinCount();
         s2 = peak2.getBinCount();
-      } else if (name == "row") {
+      } else if (name == "Row") {
         s1 = peak1.getRow();
         s2 = peak2.getRow();
-      } else if (name == "col") {
+      } else if (name == "Col") {
         s1 = peak1.getCol();
         s2 = peak2.getCol();
       }
       if (std::fabs(s1 - s2) > tolerance) {
```

Comment:
Could we add an `else` that would throw an exception with an unknown name? I think that would be an error if we get one that does not match?

Reply:
Adding a default "else" here would make the `PeaksWorkspace` a lot less flexible and potentially confusing for the users.  For example, users could add additional columns to an existing PeaksWorkspace to store some metadata or derived values.  Currently this will not affect how PeaksWorkspace function. However, if we add a default case here that will throw an error, we are basically telling the users

- You are not supposed to add any additional columns to an existing peak workspaces since the PeaksWorkspace should have a subset of predefined (allowed) columns.
- You can add new columns in if you try, and Mantid will not try to stop you unless you are trying to use `CompareWorkspaces`

Allowing users to do one thing in one location but not another will lead to confusion, and I believe Mantid should (try to) deliver a consistent messages across all algorithms.

Reply:
I must admit I hadn't checked but is it possible to add columns to a `PeaksWorkspace`?

I completely agree that if we can do that then we don't want to make this an error here. Perhaps we could issue a warning instead just so there is something rather than silently slipping through?

Reply:
Yes, `PeaksWorkspace` has a method that will allow one to add a new column to it on the fly
```
pws.addColumn?
Docstring:
addColumn( (ITableWorkspace)self, (str)type, (str)name) -> bool :
    Add a named column with the given type. Recognized types are: int,float,double,bool,str,V3D,long64

addColumn( (ITableWorkspace)self, (str)type, (str)name, (int)plottype) -> bool :
    Add a named column with the given datatype (int,float,double,bool,str,V3D,long64) 
    and plottype (0 = None, 1 = X, 2 = Y, 3 = Z, 4 = xErr, 5 = yErr, 6 = Label).
Type:      method
```

We could add a warning message, but I would recommend setting its level to information (or even debug) so that it does not flood the stdout when users run it.

Reply:
I forgot to mention that `pws` was created with
```
CreatePeaksWorkspace(NumberOfPeaks=3, OutputWorkspace='pws')
```

Reply:
Ah yes of course, it extends the table workspace interface. Thanks.

My primary aim with the column name is just to make sure that users don't think a thing is successful when actually it hasn't done what they expected, which is I think what was happening before your fixes here? Do you expect users to see this warning often? Perhaps I am misunderstanding the workflow?

Reply:
Yes, the issue was that `CompareWorkspaces` skipped all columns due to name mismatch, and users (including most developers) were assuming that all columns were compared.

As for the warning messages, if the workflow script users uses have a lot of appended columns, they will see a lot of warning messages on the terminal stating that those columns are not compared. I am inclined to have those warning set to information or debug level so that only developers and experienced users can see them while the rest regular users don't get overwhelmed by the additional information.

Reply:
I don't think too many people do this but information level is fine.